### PR TITLE
new(ts): Implementation of the last basics interactions

### PR DESCRIPTION
**Pull request by digwendal**
Adding the last single transition interaction (KeyTyped and WindowClose)

### CODE REVIEW

File: malai-core/org.malai.ts/src/interaction/Events.ts [hunk 1]
```diff
@@ -58,6 +59,10 @@ export function isTextInput(target: EventTarget): target is Element {
         && (<Element>target).getAttribute("type") === "text");
 }
 
+export function isWindowClose(event: Event): boolean {
```
Comment (arnobl) [inline malai-core/org.malai.ts/src/interaction/Events.ts]: isWindowClosed

File: malai-core/org.malai.ts/src/interaction/Events.ts [hunk 2]
```diff
@@ -18,7 +18,8 @@ export enum EventRegistrationToken {
     Click = "click",
     Input = "input",
     Scroll = "scroll",
-    Change = "change"
+    Change = "change",
+    Beforunload = "beforeunload"
```
Comment (arnobl) [inline malai-core/org.malai.ts/src/interaction/Events.ts]: BeforeUnload

File: malai-core/org.malai.ts/src/interaction/library/WindowClose.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+import {TSFSM} from "../TSFSM";
+import {FSMDataHandler} from "../FSMDataHandler";
+import {TerminalState} from "../../../src-core/fsm/TerminalState";
+import {isWindowClose} from "../Events";
+import {TSInteraction} from "../TSInteraction";
+import {InteractionData} from "../../../src-core/interaction/InteractionData";
+import {WindowCloseTransition} from "../WindowCloseTransition";
+
+export class WindowCloseFSM extends TSFSM<WindowCloseFSMHandler> {
```
Comment (arnobl) [inline malai-core/org.malai.ts/src/interaction/library/WindowClose.ts]: WindowClosedFSM

File: malai-core/org.malai.ts/src/interaction/library/WindowClose.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            this.checkButton = buttonToCheck;
+        }
+    }
+
+    public getCheckButton(): number {
+        return this.checkButton === undefined ? -1 : this.checkButton;
+    }
+}
+
+interface WindowCloseFSMHandler extends FSMDataHandler {
```
Comment (arnobl) [inline malai-core/org.malai.ts/src/interaction/library/WindowClose.ts]: WindowClosedFSMHandler

File: malai-core/org.malai.ts/src/interaction/library/WindowClose.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    initToCloseHandler(event: Event): void;
+}
+
+/**
+ * A user interaction for pressing down the mouse button.
+ * @author Gwendal DIDOT
+ */
+export class WindowClose extends TSInteraction<InteractionData, WindowCloseFSM, Node> {
```
Comment (arnobl) [inline malai-core/org.malai.ts/src/interaction/library/WindowClose.ts]: WindowClosed

File: malai-core/org.malai.ts/test/interaction/WindowClose.test.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
+/*
```
Comment (arnobl) [inline malai-core/org.malai.ts/test/interaction/WindowClose.test.ts]: Rename the file to WindowClosed

**Comment by digwendal** [comment]
Change done

**Pull request merged**